helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -97,7 +97,7 @@ public void shutdown() {
   }
 
   @VisibleForTesting
-  void shutdownNow() {
+  public void shutdownNow() {
```

Comment:
Is the VisibleForTesting not enough? This is not a good idea to expose it .

Reply:
helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkHelixJavaApis.java
actually leaks hundreds of threads.
We need to call this method to avoid the thread leak
```
public void afterClass() throws Exception {
    String testClassName = getClass().getSimpleName();
    System.out.println("AfterClass: " + testClassName + " of TestMultiZkHelixJavaApis called.");

    try {
      // Kill all mock controllers and participants
      MOCK_CONTROLLERS.values().forEach(ClusterControllerManager::syncStop);
      MOCK_PARTICIPANTS.forEach(mockParticipantManager -> {
        mockParticipantManager.syncStop();
        StateMachineEngine stateMachine = mockParticipantManager.getStateMachineEngine();
        if (stateMachine != null) {
          StateModelFactory stateModelFactory = stateMachine.getStateModelFactory("Task");
          if (stateModelFactory != null && stateModelFactory instanceof TaskStateModelFactory) {
            ((TaskStateModelFactory) stateModelFactory).shutdownNow();
          }
        }
      });

```

Reply:
JK suggestion, revert this one and let it leak for now. 

We may need another diff to let shutdown() have parameter to control if shutdown or shutdown now.